app/src/main/java/org/schabi/newpipe/player/VideoPlayerImpl.java

```diff
@@ -1565,6 +1571,20 @@ private int getStatusBarHeight() {
         return statusBarHeight;
     }
 
+    /*
+    * Compares current status bar height with default status bar height in Android and decides,
+    * does the device has cutout or not
+    * */
+    private boolean hasCutout(final float statusBarHeight) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            final DisplayMetrics metrics = getRootView().getResources().getDisplayMetrics();
+            final float defaultStatusBarHeight = TypedValue.applyDimension(
+                    TypedValue.COMPLEX_UNIT_DIP, 25, metrics);
+            return statusBarHeight > defaultStatusBarHeight;
+        }
+        return false;
+    }
+
```

Comment:
That looks like it should be in DeviceUtils.

Reply:
Force-pushed (which was a bad idea but anyway) a commit with a new place for this code